lib/core/src/nwc/mod.rs

```diff
@@ -0,0 +1,550 @@
+            }
+        };
+
+        let eb = EventBuilder::new(Kind::WalletConnectResponse, encrypted_content)
+            .tags([Tag::event(event.id), Tag::public_key(client_pubkey)]);
+        if let Err(e) = self.send_event(eb).await {
+            warn!("Could not send response event to relay pool: {e:?}");
+        }
+        info!("sent encrypted NWC response");
+    }
+
+    async fn handle_local_notification(
+        &self,
+        result: &Option<ResponseResult>,
+        error: &Option<NIP47Error>,
+        event_id: &str,
+    ) -> Result<()> {
+        info!("Handling notification: {result:?} {error:?}");
+        let event: SdkEvent = match (result, error) {
+            (Some(ResponseResult::PayInvoice(response)), None) => SdkEvent::NWC {
+                details: NwcEvent::PayInvoice {
+                    success: true,
+                    preimage: Some(response.preimage.clone()),
+                    fees_sat: response.fees_paid.map(|f| f / 1000),
+                    error: None,
+                },
+                event_id: event_id.to_string(),
+            },
+            (None, Some(error)) => match error.code {
+                ErrorCode::PaymentFailed => SdkEvent::NWC {
+                    details: NwcEvent::PayInvoice {
+                        success: false,
+                        preimage: None,
+                        fees_sat: None,
+                        error: Some(error.message.clone()),
+                    },
+                    event_id: event_id.to_string(),
+                },
+                _ => {
+                    warn!("Unhandled error code: {:?}", error.code);
+                    return Ok(());
+                }
+            },
+            (Some(ResponseResult::ListTransactions(_)), None) => SdkEvent::NWC {
+                details: NwcEvent::ListTransactions,
+                event_id: event_id.to_string(),
+            },
+            (Some(ResponseResult::GetBalance(_)), None) => SdkEvent::NWC {
+                details: NwcEvent::GetBalance,
+                event_id: event_id.to_string(),
+            },
+            _ => {
+                warn!("Unexpected combination");
+                return Ok(());
+            }
+        };
+        info!("Sending event: {event:?}");
+        self.event_manager.notify(event).await;
+        Ok(())
+    }
+
+    async fn forward_payment_to_clients(
+        &self,
+        p: &Payment,
+        clients: &HashMap<String, NostrWalletConnectURI>,
+    ) {
+        let (invoice, description, preimage, payment_hash) = match &p.details {
+            crate::model::PaymentDetails::Lightning {
+                invoice,
+                description,
+                preimage,
+                payment_hash,
+                ..
+            } => (
+                invoice.clone().unwrap_or_default(),
+                description.clone(),
+                preimage.clone().unwrap_or_default(),
+                payment_hash.clone().unwrap_or_default(),
```

Comment:
Should we forward the payment if these are not set? I imagine at least the payment_hash is important?